ugvc/mrd/balanced_strand_utils.py

```diff
@@ -0,0 +1,992 @@
+                    balanced_tags[TrimmerSegmentTags.NATIVE_ADAPTER.value] >= 1
+                    or balanced_tags[TrimmerSegmentTags.STEM_END.value] >= self.min_stem_end_matched_length
+                )
+                record.info[HistogramColumnNames.STRAND_RATIO_END.value] = np.nan
+                if not is_end_reached:
+                    record.info[
+                        HistogramColumnNames.STRAND_RATIO_CATEGORY_END.value
+                    ] = BalancedCategories.END_UNREACHED.value
+                else:
+                    if self.min_total_hmer_lengths_in_tags <= tags_sum_end <= self.max_total_hmer_lengths_in_tags:
+                        record.info[HistogramColumnNames.STRAND_RATIO_END.value] = T_hmer_end / (
+                            T_hmer_end + A_hmer_end
+                        )
+                    record.info[HistogramColumnNames.STRAND_RATIO_CATEGORY_END.value] = get_annotation(
+                        record.info[HistogramColumnNames.STRAND_RATIO_END.value],
+                        self.sr_lower,
+                        self.sr_upper,
+                    )  # this works for nan values as well - returns UNDETERMINED
+                print(
+                    is_end_reached,
+                    balanced_tags[TrimmerSegmentTags.NATIVE_ADAPTER.value],
+                    balanced_tags[TrimmerSegmentTags.STEM_END.value],
+                    tags_sum_end,
+                    (self.min_total_hmer_lengths_in_tags <= tags_sum_end <= self.max_total_hmer_lengths_in_tags),
+                    balanced_tags,
+                )
+                print(record)
+            records_out[j] = record
+
+        return records_out
+
+
+# Display defaults
+STRAND_RATIO_AXIS_LABEL = "LIG/HYB strands ratio"
+
+# Misc
+balanced_category_list = [v.value for v in BalancedCategories.__members__.values()]
+supported_adapter_versions = BalancedStrandAdapterVersions.__members__.values()
+
+
+# pylint: disable=missing-param-doc
+def _assert_adapter_version_supported(adapter_version: str | BalancedStrandAdapterVersions):
+    """
+    Assert that the adapter version is supported
+
+    Parameters
+    ----------
+    adapter_version : str | BalancedStrandAdapterVersions
+        adapter version to check
+
+    Raises
+    ------
+    AssertionError
+        If the adapter version is not supported
+    """
+    if isinstance(adapter_version, BalancedStrandAdapterVersions):
+        assert adapter_version in supported_adapter_versions, (
+            f"Unsupported adapter version {adapter_version.value}, "
+            + f"supprted values are {', '.join(supported_adapter_versions)}"
+        )
+    if isinstance(adapter_version, str):
+        assert adapter_version in [v.value for v in adapter_version], (
+            f"Unsupported adapter version {adapter_version}, "
+            + f"supprted values are {', '.join(supported_adapter_versions)}"
+        )
+
+
+def get_annotation(x, sr_lower, sr_upper):
```

Comment:
will be nice to rename x is strand_ratio.
function name is also a bit too generic.

Reply:
Done